### PR TITLE
expression: rewrite builtin function: ROUND, ABS

### PR DESCRIPTION
**Pull request by breeswish**

### CODE REVIEW

File: expression/builtin_math.go [hunk 1]
```diff
@@ -104,42 +111,262 @@ type absFunctionClass struct {
...
+	return math.Abs(val), false, nil
+}
+
+type builtinAbsIntToIntSig struct {
+	baseIntBuiltinFunc
+}
+
+// evalInt evals ABS(value).
+// See https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_abs
+func (b *builtinAbsIntToIntSig) evalInt(row []types.Datum) (int64, bool, error) {
```
Comment (zz-jason) [inline expression/builtin_math.go]: s/builtinAbsIntToIntSig/builtinAbsIntSig

File: expression/builtin_math.go [hunk 2]
```diff
@@ -104,42 +111,262 @@ type absFunctionClass struct {
...
+	return -val, false, nil
+}
+
+type builtinAbsUIntToUIntSig struct {
+	baseIntBuiltinFunc
+}
+
+// evalInt evals ABS(value).
+// See https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_abs
+func (b *builtinAbsUIntToUIntSig) evalInt(row []types.Datum) (int64, bool, error) {
```
Comment (zz-jason) [inline expression/builtin_math.go]: ditto

File: expression/builtin_math.go [hunk 3]
```diff
@@ -104,42 +111,262 @@ type absFunctionClass struct {
...
+	return b.args[0].EvalInt(row, b.ctx.GetSessionVars().StmtCtx)
+}
+
+type builtinAbsDecToDecSig struct {
+	baseDecimalBuiltinFunc
+}
+
+// evalDecimal evals ABS(value).
+// See https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_abs
+func (b *builtinAbsDecToDecSig) evalDecimal(row []types.Datum) (*types.MyDecimal, bool, error) {
```
Comment (zz-jason) [inline expression/builtin_math.go]: ditto

File: expression/builtin_math.go [hunk 4]
```diff
@@ -104,42 +111,262 @@ type absFunctionClass struct {
...
+// See https://dev.mysql.com/doc/refman/5.7/en/mathematical-functions.html#function_round
+func (b *builtinRoundRealSig) evalReal(row []types.Datum) (float64, bool, error) {
+	val, isNull, err := b.args[0].EvalReal(row, b.ctx.GetSessionVars().StmtCtx)
+	if isNull || err != nil {
+		return 0, isNull, errors.Trace(err)
+	}
+	return types.Round(val, 0), false, nil
+}
+
+type builtinRoundIntToIntSig struct {
```
Comment (zz-jason) [inline expression/builtin_math.go]: ditto

**Comment by breeswish** [comment]
@zz-jason Done

**Comment by zz-jason** [comment]
LGTM

**Pull request merged**